crates/wasi-common/src/hostcalls_impl/fs.rs

```diff
@@ -561,12 +561,16 @@ pub(crate) unsafe fn path_open(
         != 0;
 
     let fd = hostcalls_impl::path_open(resolved, read, write, oflags, fs_flags)?;
-
-    // Determine the type of the new file descriptor and which rights contradict with this type
-    let (_ty, max_base, max_inheriting) = determine_type_rights(&fd)?;
     let mut fe = FdEntry::from(fd)?;
-    fe.rights_base &= max_base;
-    fe.rights_inheriting &= max_inheriting;
+    // We need to manually deny the rights which are not parent's inheriting rights.
+    // This should not be needed, but currently determine_type_and_access_rights,
+    // which is used by FdEntry::from, may grant more rights than requested.
+    //
+    // This is a hotfix. It needs to be discussed whether or not `path_open`
```

Comment:
Let's have that discussion here so we don't add a comment with `this is a hotfix`.

I think @sunfishcode is more qualified to answer this than I.  On the surface it sounds wrong to me, though.

Reply:
Perhaps we could change "This is a hotfix." for "TODO"?

Reply:
I intend to remove the _this needs to be discussed_ part when we've discussed it :)

Reply:
I intend to remove the _it needs to be discussed_ part when we'll have discussed  it ;)